shared/devices/add-device/container.js

```diff
@@ -30,6 +51,10 @@ export default Container.namedConnect<
 >(
   () => ({}),
   mapDispatchToProps,
-  (s, d, o) => ({...d, highlight: o.routeProps.get('highlight'), title: 'Add a device'}),
+  (s, d, o) => ({
+    ...d,
+    highlight: flags.useNewRouter ? o.navigation.getParam('highlight', []) : o.routeProps.get('highlight'),
```

Comment:
i know i was doing this before to get nav2 merged in but maybe we should make a helper in util/container that'll get routeprops generically so we can just
`Container.getProps(ownProps, key)`